vpr/src/route/four_ary_heap.h

```diff
@@ -0,0 +1,18 @@
+#ifndef VTR_FOUR_ARY_HEAP_H
+#define VTR_FOUR_ARY_HEAP_H
+
+#include "k_ary_heap.h"
+#include <vector>
+
+class FourAryHeap : public KAryHeap {
```

Comment:
Add Doxygen comments for the whole class.  If not covered by the base class .h file, also add a Doxygen comment for these methods (but if the base class totally covers them that's OK).
Quote the results in here on speed of this 4-ary heap vs. a binary heap.